src/main/java/com/unbxd/SchemaManager/dao/MongoDao.java

```diff
@@ -0,0 +1,95 @@
+import org.bson.conversions.Bson;
+import java.util.ArrayList;
+import static org.bson.codecs.configuration.CodecRegistries.fromProviders;
+import static org.bson.codecs.configuration.CodecRegistries.fromRegistries;
+
+@Log4j2
+public class MongoDao implements SchemaDao {
+
+    private MongoDatabase database;
+
+    public MongoDao(String host,int port,String dbname){
+        MongoClientURI mongoClientURI = new MongoClientURI("mongodb://"+host+":"+port);
+        log.info("Connecting to mongo with with using host: "+host+" and port: "+port);
+        CodecRegistry pojoCodecRegistry = fromRegistries(MongoClientSettings.getDefaultCodecRegistry(),
+                fromProviders(PojoCodecProvider.builder().register(Field.class).register(SiteSchema.class).automatic(true).build()));
+        MongoClient mongoClient  = new MongoClient(mongoClientURI);
+        this.database = mongoClient.getDatabase(dbname).withCodecRegistry(pojoCodecRegistry);
+    }
+
+    @Override
+    public void addNewSchema(SiteSchema schema) throws DaoException{
+        try {
+            database.createCollection(schema.getSiteKey());
+            MongoCollection<SiteSchema> collection = database.getCollection(schema.getSiteKey(), SiteSchema.class);
+            collection.insertOne(schema);
+        }
+        catch (Exception e){
+            throw new DaoException(500,"Site already present, "+e.getLocalizedMessage());
+        }
+    }
+
+    @Override
+    public SiteSchema getSchemaForSite(String siteKey) throws DaoException{
+        try {
+            MongoCollection<SiteSchema> collection = database.getCollection(siteKey,SiteSchema.class);
+            SiteSchema res = collection.find().first();
+            return res;
+        }
+        catch (Exception e){
+            throw new DaoException(500,e.getLocalizedMessage());
+        }
+    }
+
+    @Override
+    public void updateSchemaForSite(String siteKey,SiteSchema schema) throws DaoException{
+        try {
+            deleteSchemaForSite(siteKey);
+            database.createCollection(siteKey);
+            MongoCollection<SiteSchema> collection = database.getCollection(siteKey,SiteSchema.class);
+            collection.insertOne(schema);
+        }
+        catch (Exception e){
+            throw new DaoException(500,e.getLocalizedMessage());
+        }
+    }
+
+    @Override
+    public void updateFieldInSite(String siteKey, String fieldName, Field field) throws DaoException {
+        try {
+            MongoCollection<SiteSchema> collection = database.getCollection(siteKey,SiteSchema.class);
+            ArrayList<Bson> updates = new ArrayList<Bson>();
+            updates.add(Updates.pull("fields", new Document("fieldname", fieldName)));
+            updates.add(Updates.push("fields", field));
+            collection.updateOne(Filters.eq("siteKey", siteKey), updates);
+        }
+        catch (Exception e) {
+            throw new DaoException(500,e.getLocalizedMessage());
```

Comment:
Would defining my own status code or error codes be wrong since i am giving a already established HTTP response code a different error message. So If I define 502 as schema field not found, would it not be confusing as 502 is generally used to mean something else